homeassistant/config_entries.py

```diff
@@ -2021,9 +2016,15 @@ async def support_remove_from_device(hass: HomeAssistant, domain: str) -> bool:
     return hasattr(component, "async_remove_config_entry_device")
 
 
-async def _load_integration(
+async def _async_get_flow_handler(
     hass: HomeAssistant, domain: str, hass_config: ConfigType
-) -> None:
+) -> type[ConfigFlow]:
+    """Get a flow handler for specified domain."""
+
+    # First check if there is a handler registered for the domain
+    if domain in hass.config.components and (handler := HANDLERS.get(domain)):
```

Comment:
I wonder if we can cache it even harder. Something might be discovered but never set up, in which case it would not benefit from this optimization. The handler returned from `_async_get_flow_handler` will always be the same, so we should be able to just cache this?

Reply:
That would be nice, but since it requires passing `hass_config` currently we end up with more stack depth anyways since we can't use `lru_cache` as its not immutable.

Will see if we really need `hass_config` here ....

Reply:
It gets passed down as `async_setup_component(hass, dep, hass_config)` so we can't get away from passing it

Reply:
Rearranging it would save one level of stack depth but than we have to pay it anyways because the lru_cache would need to to be another function.